bindings/http/http.go

```diff
@@ -81,6 +107,59 @@ func (h *HTTPSource) Init(metadata bindings.Metadata) error {
 	return nil
 }
 
+// readMTLSCertificates reads the certificates and key from the metadata and returns a tls.Config.
+func (h *HTTPSource) readMTLSCertificates() (*tls.Config, error) {
+	if h.metadata.MTLSClientCert == "" || h.metadata.MTLSClientKey == "" || h.metadata.MTLSRootCA == "" {
+		return nil, fmt.Errorf("metadata %q is set to %s but required certificates and key are missing", MTLSEnable, h.metadata.MTLSEnable)
+	}
+	caCertBytes, err := h.getPemBytes(MTLSRootCA, h.metadata.MTLSRootCA)
+	if err != nil {
+		return nil, err
+	}
+	clientCertBytes, err := h.getPemBytes(MTLSClientCert, h.metadata.MTLSClientCert)
+	if err != nil {
+		return nil, err
+	}
+	clientKeyBytes, err := h.getPemBytes(MTLSClientKey, h.metadata.MTLSClientKey)
+	if err != nil {
+		return nil, err
+	}
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCertBytes)
+	cert, err := tls.X509KeyPair(clientCertBytes, clientKeyBytes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load client certificate: %w", err)
+	}
+	return &tls.Config{
+		MinVersion:   tls.VersionTLS12,
+		RootCAs:      caCertPool,
+		Certificates: []tls.Certificate{cert},
+	}, nil
+}
+
+// getPemBytes returns the PEM encoded bytes from the provided certName and certData.
+// If the certData is a file path, it reads the file and returns the bytes.
+// Else if the certData is a PEM encoded string, it returns the bytes.
+// Else it returns an error.
+func (h *HTTPSource) getPemBytes(certName, certData string) ([]byte, error) {
+	// Read the file
+	pemBytes, err := os.ReadFile(certData)
+	// If there is an error assume it is already PEM encoded string not a file path.
+	if err != nil {
```

Comment:
Check if error is specifically `errors.Is(err, os.ErrNotExist)` to exclude other errors such as permissions ones